Integrate heterogenous auto planner to TGIF DI

gnahzg opened the pull request:
Summary:
As titled

Integrate TorchRec autoplanner for uneven sharding through option use_trec_hetero_auto_planner. The autoplanner would only be used when user explicitly set so.

Continue testing would be done before we fully migrate to use TorchRec Hetero Autoplanner. Aiming for timeline end of H1 2024

Differential Revision: D56203531

jingsh reviewed (approved):
lgtm

could you fix the pyre errors?

gnahzg committed:
Integrate heterogenous auto planner to TGIF DI (#1978)

Summary:

As titled

Integrate TorchRec autoplanner for uneven sharding through option use_trec_hetero_auto_planner. The autoplanner would only be used when user explicitly set so.

Continue testing would be done before we fully migrate to use TorchRec Hetero Autoplanner. Aiming for timeline end of H1 2024

Reviewed By: jingsh

Differential Revision: D56203531
